### ISSUE TITLE
Request Help:  Error in implementing open-source in HTMLRAG

### ISSUE THREAD

**Issue by dev-wenjie** [plageon/HtmlRAG#4]
Hello, I am Yin.
- I used your open-source to build environment.

1. Installation
2. Dependencies
3. data preparation
--------successfully donn by your manual

4. HTML cleaning
 . Executed the command in this step
  `bash ./scripts/simplify_html.sh` : error occured like below
  
`Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Python312\Lib\multiprocessing\spawn.py", line 122, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\multiprocessing\spawn.py", line 132, in _main
    self = reduction.pickle.load(from_parent)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: Can't get attribute 'simplify_html_lines' on <module '__mp_main__' from 'soucre_directory_path\\html4rag\\simplify_html.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Python312\Lib\multiprocessing\spawn.py", line 122, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\multiprocessing\spawn.py", line 132, in _main
    self = reduction.pickle.load(from_parent)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: Can't get attribute 'simplify_html_lines' on <module '__mp_main__' from 'soucre_directory_path\\html4rag\\simplify_html.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Python312\Lib\multiprocessing\spawn.py", line 122, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\multiprocessing\spawn.py", line 132, in _main
    self = reduction.pickle.load(from_parent)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: Can't get attribute 'simplify_html_lines' on <module '__mp_main__' from 'soucre_directory_path\\html4rag\\simplify_html.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Python312\Lib\multiprocessing\spawn.py", line 122, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\multiprocessing\spawn.py", line 132, in _main
    self = reduction.pickle.load(from_parent)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: Can't get attribute 'simplify_html_lines' on <module '__mp_main__' from 'soucre_directory_path\\html4rag\\simplify_html.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Python312\Lib\multiprocessing\spawn.py", line 122, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\multiprocessing\spawn.py", line 132, in _main
    self = reduction.pickle.load(from_parent)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: Can't get attribute 'simplify_html_lines' on <module '__mp_main__' from 'soucre_directory_path\\html4rag\\simplify_html.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Python312\Lib\multiprocessing\spawn.py", line 122, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\multiprocessing\spawn.py", line 132, in _main
    self = reduction.pickle.load(from_parent)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: Can't get attribute 'simplify_html_lines' on <module '__mp_main__' from 'soucre_directory_path\\html4rag\\simplify_html.
2024-11-20 11:44:35.042 | INFO     | __main__:<module>:57 - All done
`
  Please analyse the error and give me solution.
  
  ------------I have extra questions---------------------
  1. operating system of this module: windows , Linux ,MacOS or anyother?
  2.  Is it possible to run HTML RAG system with only this open source?
      If not, is there any something that used like cloud services?
  3.  I just followed your read.me manual. Is there any other steps to needed?
       Read,me file is updated? or, Is there any othe problem that can be happen to use this open source.
       
       Best, Regards.

**Comment by plageon** [comment]
It seems your mutiprocessing module behaves a bit wierd... Maybe you could modify the script here and directly call the function `simplify_html_lines` in https://github.com/plageon/HtmlRAG/blob/main/html4rag/simplify_html.py.

**Comment by dev-wenjie** [comment]
Thanks plageon.
I resolved the above mentioned issue.
But new problem is occurred.
It's about data preparation.
[Download datasets used in the paper]Step:
There is table that is discribe the type of data.
I couldn't get the html_sample and didn't find the method.
In there, how can I get html_sample data?
I would appreciate it if you could teach me how to get the html_sample.

**Comment by plageon** [comment]
The full processed data is available in huggingface dataset [HtmlRAG-test](https://huggingface.co/datasets/zstanjj/HtmlRAG-test).

**Comment by dev-wenjie** [comment]
Thanks for reply.
Ok I see but how should I use [Hugging Face]?
Please let me know manual.
Thanks.

**Comment by plageon** [comment]
Refer to official docs [here](https://huggingface.co/docs).

**Comment by dev-wenjie** [comment]
Hi, I am yin.
Thanks for your reply.
I have one qustion.
How should I get test html data(html sample) on Hugging Face?
Let me send if you have the sample data.
Best regards.
________________________________

**Comment by dev-wenjie** [comment]
I am sorry, it's urgent.
So I need your help.
Please let me send the sample data if you have regarding
ASQA    HotpotQA        NQ      TriviaQA        MuSiQue ELI5
.
Thanks.
________________________________
